cmd/ocm/gcp/create-wif-config.go

```diff
@@ -0,0 +1,369 @@
+	projectId := wifOutput.Spec.ProjectId
+	fmtRoleResourceId := func(role models.Role) string {
+		return fmt.Sprintf("roles/%s", role.Id)
+	}
+
+	// Create service accounts
+	for _, serviceAccount := range wifOutput.Status.ServiceAccounts {
+		serviceAccountID := serviceAccount.GetId()
+		serviceAccountName := wifOutput.Spec.DisplayName + "-" + serviceAccountID
+		serviceAccountDesc := poolDescription + " for WIF config " + wifOutput.Spec.DisplayName
+
+		fmt.Println("Creating service account", serviceAccountID)
+		_, err := CreateServiceAccount(gcpClient, serviceAccountID, serviceAccountName, serviceAccountDesc, projectId, true)
+		if err != nil {
+			return errors.Wrap(err, "Failed to create IAM service account")
+		}
+		log.Printf("IAM service account %s created", serviceAccountID)
+	}
+
+	// Create roles that aren't predefined
+	for _, serviceAccount := range wifOutput.Status.ServiceAccounts {
+		for _, role := range serviceAccount.Roles {
+			if role.Predefined {
+				continue
+			}
+			roleID := role.Id
+			roleName := role.Id
+			permissions := role.Permissions
+			existingRole, err := GetRole(gcpClient, roleID, projectId)
+			if err != nil {
+				if gerr, ok := err.(*googleapi.Error); ok && gerr.Code == 404 &&
+					strings.Contains(gerr.Message, "Requested entity was not found") {
+					existingRole, err = CreateRole(gcpClient, permissions, roleName,
+						roleID, roleDescription, projectId)
+					if err != nil {
+						return errors.Wrap(err, fmt.Sprintf("Failed to create %s", roleName))
+					}
+					log.Printf("Role %s created", roleID)
+				} else {
+					return errors.Wrap(err, "Failed to check if role exists")
+				}
+			}
+			// Update role if permissions have changed
+			if !reflect.DeepEqual(existingRole.IncludedPermissions, permissions) {
+				existingRole.IncludedPermissions = permissions
+				_, err := UpdateRole(gcpClient, existingRole, roleName)
+				if err != nil {
+					return errors.Wrap(err, fmt.Sprintf("Failed to update %s", roleName))
+				}
+				log.Printf("Role %s updated", roleID)
+			}
+		}
+	}
+
+	// Bind roles and grant access
+	for _, serviceAccount := range wifOutput.Status.ServiceAccounts {
+		serviceAccountID := serviceAccount.GetId()
+
+		fmt.Printf("\t\tBinding roles to %s\n", serviceAccount.Id)
+		roles := make([]string, 0, len(serviceAccount.Roles))
+		for _, role := range serviceAccount.Roles {
+			if !role.Predefined {
+				fmt.Printf("Skipping role %q for service account %q as custom roles are not yet supported.",
+					role.Id, serviceAccount.Id)
+				continue
+			}
+			roles = append(roles, fmtRoleResourceId(role))
+		}
+		member := fmt.Sprintf("serviceAccount:%s@%s.iam.gserviceaccount.com", serviceAccountID, projectId)
+		err := EnsurePolicyBindingsForProject(gcpClient, roles, member, projectId)
+		if err != nil {
+			log.Fatalf("Failed to bind roles to service account %s: %s", serviceAccountID, err)
```

Comment:
Pls review all  log.Fatalf stmts,  I think errors in all methods/functions ( except for the top level cmd function) should just be surfaced up /returned and let the caller chose what action to do. I dont think we should just. os.Exit(1) anywhere in the codepath